assets/js/base/components/quantity-selector/index.tsx

```diff
@@ -78,19 +81,25 @@ const QuantitySelector = ( {
 				className="wc-block-components-quantity-selector__input"
 				disabled={ disabled }
 				type="number"
-				step="1"
-				min="0"
+				step={ step }
+				min={ minimum }
 				value={ quantity }
 				onKeyDown={ quantityInputOnKeyDown }
 				onChange={ ( event ) => {
-					let value =
-						! isNumber( event.target.value ) || ! event.target.value
-							? 0
-							: parseInt( event.target.value, 10 );
+
+					let valueIsNumber = event.target.value && ( isNumber( event.target.value ) || parseInt( event.target.value, 10 ).toString() === event.target.value );
```

Comment:
```
isNumber( event.target.value )
```

This is probably not needed -- I believe there was a bug here as `event.target.value` is always `string`, not `number`.

If you do a quick test on `trunk` you'll notice that every time you enter a value inside the input field, it always gets converted to `1`.

Reply:
Parsing and then calling `toString` doesn't make sense to me here,  are trying to test if calling `parseInt` on a non-number would return `NaN` which you would then parse back to compare

What exactly is the point of this logic?

I'd be lying if I said the original was as obvious 😅 4 lines of codes has been changed by 4 different people.

> I believe isNumber( event.target.value ) was incorrectly used here as event.target.value is always string, not number.

I agree, I believe this is a regression in [this PR](https://github.com/woocommerce/woocommerce-gutenberg-products-block/pull/4088) @opr 

@franticpsyx before expanding on this function logic, I'd prefer we tame it down a bit. This can be first fixing the `isNumber` issue (by using `isNaN` instead).
@opr is the `isNumber` call needed here? switching to `isNaN( parseInt( event.target.value, 10 ) )` doesn't throw any typescript errors for me. It seems `isNaN is typed wrongly in that it only accepts `number` while it should accept everything.

@franticpsyx Would this logic still work? What I go from your changes is that you're trying to use `quantity` instead of `0`, in that case, this should be enough?

```js
let value = parseInt( event.target.value, 10 );
value =
	isNaN( value ) || ! event.target.value
		? quantity
		: value;
```

Reply:
Yes I agree, we shouldn't be using `isNumber` here after all. `isNaN` works 👍🏼

Reply:
`isNaN` should be fine.

`parseInt( event.target.value, 10 ).toString() === event.target.value` is just a more verbose attempt to do the same, while only accepting integer values. As far as I could tell, the codebase in its current form was not designed to handle floats (?) all the way from that API to the front-end.